packages/xo-web/src/common/editable/index.js

```diff
@@ -343,68 +340,65 @@ export class Number extends Component {
 
 export class Select extends Editable {
   static propTypes = {
-    options: PropTypes.oneOfType([PropTypes.array, PropTypes.object])
-      .isRequired,
     renderer: PropTypes.func,
   }
 
-  componentWillReceiveProps (props) {
-    if (
-      props.value !== this.props.value ||
-      props.options !== this.props.options
-    ) {
-      this.setState({
-        valueKey: findKey(props.options, option => option === props.value),
-      })
-    }
-  }
-
-  get value () {
-    return this.props.options[this.state.valueKey]
+  _closeEditionIfUnfocused = () => {
+    this._focused = false
+    setTimeout(() => {
+      !this._focused && this._closeEdition()
+    }, 10)
   }
 
-  _onChange = event => {
-    this.setState({ valueKey: getEventValue(event) }, this._save)
+  _focus = () => {
+    this._focused = true
   }
 
-  _optionToJsx = (option, key) => {
-    const { renderer } = this.props
-
+  _renderDisplay () {
+    const { children, renderer, value } = this.props
     return (
-      <option key={key} value={key}>
-        {renderer ? renderer(option) : option}
-      </option>
+      children || (
+        <span>
+          {renderer !== undefined ? renderer(value) : this._renderer()}
+        </span>
+      )
     )
   }
 
-  _onEditionMount = ref => {
-    // Seems to work in Google Chrome (not in Firefox)
-    ref && ref.dispatchEvent(new window.MouseEvent('mousedown'))
-  }
+  _renderer = () => {
+    const { value, multi } = this.props
+    if (value === undefined || (multi && isEmpty(value))) {
+      return _('noValue')
+    }
 
-  _renderDisplay () {
-    const { children, renderer, value } = this.props
+    return !multi ? value : value.map(v => v.label).join(', ')
+  }
 
-    return children || <span>{renderer ? renderer(value) : value}</span>
+  _selectAll = () => {
+    const { onChange, options } = this.props
+    return onChange(options)
   }
 
   _renderEdition () {
-    const { saving, valueKey } = this.state
-    const { options } = this.props
+    const { props } = this
 
     return (
-      <select
-        autoFocus
-        className={classNames('form-control', styles.select)}
-        onBlur={this._closeEdition}
-        onChange={this._onChange}
-        onKeyDown={this._onKeyDown}
-        readOnly={saving}
-        ref={this._onEditionMount}
-        value={valueKey}
+      <form
+        className='input-group'
+        onBlur={this._closeEditionIfUnfocused}
+        onFocus={this._focus}
       >
-        {map(options, this._optionToJsx)}
-      </select>
+        <ReactSelect {...props} openOnFocus onKeyDown={this._onKeyDown} />
```

Comment:
`props` contains `options` that can be an object. `ReactSelect` doesn't support an object. IMO, we shouldn't accept an object so add the correct prop types and fix the bad usages.